src/code/z_camera_data.c

```diff
@@ -0,0 +1,3774 @@
+#include "ultra64.h"
```

Comment:
Because this is such a large file, I think it would be good from a user perspective to have some sort of table of contents at the top, with an explanation of how this ragged array stuff with the validity checks and so on works. Seems like it goes broadly
- individual mode data, sorted by setting
- arrays for each setting pointing to the individual data
- overall array with validity checks
- update function stuff
- misc
?